File: app/src/main/java/com/flatstack/android/main_screen/TestDialog.java
package com.flatstack.android.main_screen;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.FragmentManager;
import android.view.View;
import android.widget.TextView;

import com.flatstack.android.R;
import com.flatstack.android.utils.ui.BaseDialogFragment;

import butterknife.Bind;

public class TestDialog extends BaseDialogFragment {

    private static final String KEY_TITLE   = "dialogTitle";
    private static final String KEY_MESSAGE = "dialogMessage";

    @Bind(R.id.dialog_title)   TextView uiTitle;
    @Bind(R.id.dialog_message) TextView uiMessage;

    private String title;
    private String message;

    @Override public int getLayoutRes() {
        return R.layout.dialog_test;
    }

    public static void show(@Nullable String title,
                            @Nullable String message,
                            @NonNull FragmentManager fm) {

        TestDialog dialog = new TestDialog();

        Bundle args = new Bundle();
        args.putString(KEY_TITLE, title != null ? title : "");
        args.putString(KEY_MESSAGE, message != null ? message : "");
        dialog.setArguments(args);

        dialog.show(fm, TestDialog.class.getName());
    }

    @Override protected void parseArguments(@NonNull Bundle args) {
        title = getArguments().getString(KEY_TITLE);
        message = getArguments().getString(KEY_MESSAGE);
    }

    @Override public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        uiTitle.setText(title != null ? title : "");
        uiMessage.setText(message != null ? message : "");
    }
}

File: app/src/main/java/com/flatstack/android/main_screen/TestDialog.kt
package com.flatstack.android.main_screen

import android.os.Bundle
import android.support.v4.app.FragmentManager
import android.view.View
import android.widget.TextView
import butterknife.BindView

import com.flatstack.android.R
import com.flatstack.android.utils.ui.BaseDialogFragment

class TestDialog : BaseDialogFragment() {
    @BindView(R.id.dialog_title) internal lateinit var uiTitle: TextView
    @BindView(R.id.dialog_message) internal lateinit var uiMessage: TextView

    private var title: String? = null
    private var message: String? = null

    override val layoutRes: Int
        get() = R.layout.dialog_test

    override fun parseArguments(args: Bundle) {
        arguments?.let {
            title = it.getString(KEY_TITLE)
            message = it.getString(KEY_MESSAGE)
        }
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        uiTitle.text = title
        uiMessage.text = message
    }

    companion object {

        private const val KEY_TITLE = "dialogTitle"
        private const val KEY_MESSAGE = "dialogMessage"

        fun show(title: String?,
                 message: String?,
                 fm: FragmentManager) {

            val dialog = TestDialog()

            val args = Bundle()
            args.putString(KEY_TITLE, title ?: "")
            args.putString(KEY_MESSAGE, message ?: "")
            dialog.arguments = args

            dialog.show(fm, TestDialog::class.java.name)
        }
    }
}
